src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -111,10 +132,37 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
                 throw new ArgumentNullException(nameof(controllerContext));
             }
 
+            var valueProvider = await CompositeValueProvider.CreateAsync(
+                controllerContext,
+                controllerContext.ValueProviderFactories);
+
+            return await BindModelAsync(parameter, controllerContext, valueProvider);
+        }
+
+        public async Task<ModelBindingResult?> BindModelAsync(
+            ParameterDescriptor parameter,
+            ControllerContext controllerContext,
+            IValueProvider valueProvider)
```

Comment:
> that's another change.

Ah well :ok:

> Lazy property doesn't work 

Yeah, I meant a memoized `async` method.

Also didn't mean "expecting" since I knew the general direction of this PR :no_bell: